### PR TITLE
Fix parse_time function

### LINKED ISSUE #1959: Function name and parameter order on "parsetime"
When using the "parsetime" function in FT.AGGREGATE, an unknown error is returned

Several things occur here:

### Typo
The name of the function does not follow the doc (or the opposite)
```parsetime``` in the doc
```parse_time```  in the code

### Parameters inversion
In the doc: ```parsetime(timesharing, [fmt]```
In the code: ```parse_time(fmt, timesharing)``` 

Notice that fmt is supposed to be optional. And it can only be optional if this is the last parameter.

Note that it will probably be solve when the parameters will be back in the right order.

Reproduction:
``` 
FT.CREATE idx:test ON hash PREFIX 1 "ktest:" SCHEMA name TEXT date TAG SORTABLE
HMSET "ktest:k1" date "20210401"
``` 

Fail 1:
```
127.0.0.1:6379> FT.AGGREGATE idx:test * APPLY 'parsetime(@date, "%Y%m%d")'
(error) Unknown function name 'parsetime'
``` 

Fail 2 (parse_time returns ```(nil)```):
```
 FT.AGGREGATE idx:test * LOAD 1 @date APPLY 'parse_time(@date, "%Y%m%d")'
1) (integer) 1
2) 1) "date"
   2) "20210401"
   3) "parse_time(@date, \"%Y%m%d\")"
   4) (nil)
```

Works:
``` 
127.0.0.1:6379> FT.AGGREGATE idx:test * APPLY 'parse_time("%Y%m%d", @date)'
1) (integer) 1
2) 1) "date"
   2) "20210401"
   3) "parse_time(\"%Y%m%d\", @date)"
   4) "1617235200"
``` 

###  Suggested fix:
- Rename the function ```parse_time``` to ```parsetime``` in the code
- Restore the normal parameter order : ```parsetime(timesharing, [fmt])```

### PR DESCRIPTION
**Pull request by ashtul**
`parse_time` function can be used.
@emmanuelkeller please note the function expects a string since it uses `strptime` behind the scene. 

fix #1959

### CODE REVIEW

**Review by emmanuelkeller — CHANGES_REQUESTED** [review]
The parameters are back in the right order , thanks!
However the name of the function is still "parse_time"
Can we rename it "parsetime" ?

File: tests/pytests/test.py [hunk 1]
```diff
@@ -2742,22 +2742,22 @@ def testMonthOfYear(env):
...
 
-    err = env.cmd('ft.aggregate', 'idx', '@test:[0..inf]', 'LOAD', '1', '@test', 'APPLY', 'parse_time(11,22)', 'as', 'a')[1]
+    err = env.cmd('ft.aggregate', 'idx', '*', 'LOAD', '1', '@test', 'APPLY', 'parse_time(11,22)', 'as', 'a')[1]
     assertEqualIgnoreCluster(env, type(err[0]), redis.exceptions.ResponseError)
 
-    env.expect('ft.aggregate', 'idx', '@test:[0..inf]', 'LOAD', '1', '@test', 'APPLY', 'parse_time("%s", "%s")' % ('d' * 2048, 'd' * 2048), 'as', 'a').equal([1L, ['test', '12234556', 'a', None]])
-
-    env.expect('ft.aggregate', 'idx', '@test:[0..inf]', 'LOAD', '1', '@test', 'APPLY', 'parse_time("test", "%s")' % ('d' * 2048), 'as', 'a').equal([1L, ['test', '12234556', 'a', None]])
+    # valid test
+    env.expect('ft.aggregate', 'idx', '*', 'LOAD', '1', '@test', 'APPLY', 'parse_time(@test, "%Y%m%d")', 'as', 'a').equal([1L, ['test', '20210401', 'a', '1617235200']])
```
Comment (emmanuelkeller) [inline tests/pytests/test.py:2760]: Do we also test when no FMT parameter is given ?

**Review by emmanuelkeller — CHANGES_REQUESTED** [review]

**Comment by emmanuelkeller** [comment]
> `parse_time` function can be used.
> @emmanuelkeller please note the function expects a string since it uses `strptime` behind the scene.
> 
> fix #1959

In the documentation we link to this page for both timefmt and parsetime. Is it still right ?
https://strftime.org

File: src/aggregate/functions/date.c [hunk 2]
```diff
@@ -272,19 +272,19 @@ static int parseTime(ExprEval *ctx, RSValue *result, RSValue **argv, size_t argc
 
   char fmtbuf[1024] = {0};
   char valbuf[1024] = {0};
-  size_t fmtlen;
 
-  const char *origfmt = RSValue_StringPtrLen(argv[0], &fmtlen);
+  size_t fmtlen;
```
Comment (rafie) [inline src/aggregate/functions/date.c:276]: better init this to 0

File: src/aggregate/functions/date.c [hunk 3]
```diff
@@ -272,19 +272,19 @@ static int parseTime(ExprEval *ctx, RSValue *result, RSValue **argv, size_t argc
...
+  const char *origfmt = RSValue_StringPtrLen(argv[1], &fmtlen);
   if (fmtlen > sizeof(fmtbuf)) {
     goto err;
   }
+  memcpy(fmtbuf, origfmt, fmtlen);
```
Comment (rafie) [inline src/aggregate/functions/date.c:281]: don't we skip the '\0` here?

**Review by emmanuelkeller — APPROVED** [review]
LGTM !

**Comment by ashtul** [comment]
> > `parse_time` function can be used.
> 
> > @emmanuelkeller please note the function expects a string since it uses `strptime` behind the scene.
> 
> > 
> 
> > fix #1959
> 
> 
> 
> @ashtul In the documentation we link to this page for both timefmt and parsetime. Is it still right ?
> 
> https://strftime.org

I believe they both have the same syntax.

File: tests/pytests/test.py (hunk 1 above)
Comment (ashtul) [reply tests/pytests/test.py]: We have it in prior test

File: src/aggregate/functions/date.c (hunk 2 above)
Comment (ashtul) [reply src/aggregate/functions/date.c:276]: `fmtlen` is used as an out param for `RSValue_StringPtrLen`.
If it fails, we exit the function so there is no meaning to initializing it.

File: src/aggregate/functions/date.c (hunk 3 above)
Comment (ashtul) [reply src/aggregate/functions/date.c:281]: This was existing code that was just moved to make it more readable.

**Pull request merged**